moveit_ros/moveit_servo/src/realtime.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 PickNik Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "moveit_servo/realtime.hpp"
+
+#include <sched.h>
+
+#include <cstring>
+#include <fstream>
+
+namespace moveit_servo
+{
+bool has_realtime_kernel()
+{
+  std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
+  bool has_realtime = false;
+  if (realtime_file.is_open())
+  {
+    realtime_file >> has_realtime;
+  }
+  return has_realtime;
+}
+
+bool configure_sched_fifo(int priority)
+{
+  struct sched_param schedp;
+  memset(&schedp, 0, sizeof(schedp));
+  schedp.sched_priority = priority;
+  return !sched_setscheduler(0, SCHED_FIFO, &schedp);
+}
```

Comment:
Please don't copy-paste this. I put these in functions so you can just call them from other places. It is just fine to depend on and link to ros2_control::controller_manager from here.

Reply:
Good idea. Do we want to link against the `controller_manager` library when we just need this little bit of functionality, though? It would be nice if `realtime` were in its own library.

Anyway, done